File: cloudflare/systems/go/wasm/shared/hello.go
//+ build js,wasm

package shared

import (
	"encoding/json"
	"syscall/js"
)

type MessageRes struct {
	Message string `json:"message"`
}

func Index(_this js.Value, args []js.Value) interface{} {
	callback := args[0]
	res := MessageRes{Message: "Auth API"}
	res_bytes, err := json.Marshal(res)
	if err != nil {
		return callback.Invoke(err.Error(), nil).String()
	}
	return callback.Invoke(nil, string(res_bytes)).String()
}

func Hello(_this js.Value, args []js.Value) interface{} {
	callback := args[0]
	res := MessageRes{Message: "hello world"}
	res_bytes, err := json.Marshal(res)
	if err != nil {
		return callback.Invoke(err.Error(), nil).String()
	}
	return callback.Invoke(nil, string(res_bytes)).String()
}

File: cloudflare/systems/wasm/src/shared/hello.rs
use serde::{Serialize, Deserialize};

#[derive(Serialize, Deserialize)]
pub struct MessageRes {
    message: String,
}

pub fn index() -> MessageRes {
    let message = MessageRes {
        message: "Auth API".to_string(),
    };
    return message;
}

pub fn hello() -> MessageRes {
    let message = MessageRes {
        message: "hello world".to_string(),
    };
    return message;
}
